ToolBox/Polygon/PolygonDataQueueHandler.cs

```diff
@@ -507,105 +508,235 @@ private IEnumerable<Tick> GetCryptoTradeTicks(HistoryRequest request)
             }
         }
 
-
         private IEnumerable<Tick> GetEquityQuoteTicks(HistoryRequest request)
         {
             // https://api.polygon.io/v2/ticks/stocks/nbbo/SPY/2020-08-24?apiKey=
 
+            bool isException = false;
+
             var start = request.StartTimeUtc;
             var end = request.EndTimeUtc;
+            var offsetStart = Time.DateTimeToUnixTimeStampNanoseconds(start);
+            var offsetEnd = Time.DateTimeToUnixTimeStampNanoseconds(end);
 
-            while (start <= end)
+            // Continues until offsetStart gets to the end of the set time at offsetEnd
+            while (offsetStart < offsetEnd)
             {
+                // Skips Saturday and Sunday
+                if (!start.IsCommonBusinessDay())
+                {
+#if DEBUG
+                    Log.Trace("Weekends are not processed " + start.ToString(System.Globalization.CultureInfo.GetCultureInfo("en-US")));
+#endif
+                    start = start.Date.AddDays(1);
+                    break;
+                }
+                else
+                {
+#if DEBUG
+                    Log.Trace("Processing " + start.ToString(System.Globalization.CultureInfo.GetCultureInfo("en-US")));
+#endif
+                }
+
                 using (var client = new WebClient())
                 {
-                    var offset = Time.DateTimeToUnixTimeStampNanoseconds(start);
-                    var url = $"{HistoryBaseUrl}/v2/ticks/stocks/nbbo/{request.Symbol.Value}/{start.Date:yyyy-MM-dd}?apiKey={_apiKey}&timestamp={offset}";
+                    var url = $"{HistoryBaseUrl}/v2/ticks/stocks/nbbo/{request.Symbol.Value}/{start.Date:yyyy-MM-dd}?apiKey={_apiKey}&timestamp={offsetStart}&limit={_apiResultsLimit}";
 
-                    var response = client.DownloadString(url);
+#if DEBUG
+                    Log.Trace(url);
+#endif
 
-                    var obj = JObject.Parse(response);
-                    var objTicks = obj["results"];
-                    if (objTicks.Type == JTokenType.Null || !objTicks.Any())
+                    string response = "";
+                    try
                     {
-                        // current date finished, move to next day
-                        start = start.Date.AddDays(1);
-                        continue;
+                        response = client.DownloadString(url);
                     }
-
-                    foreach (var objTick in objTicks)
+                    catch (WebException e)
                     {
-                        var row = objTick.ToObject<EquityQuoteTickResponse>();
-
-                        var utcTime = Time.UnixNanosecondTimeStampToDateTime(row.ExchangeTimestamp);
+                        isException = true;
```

Comment:
Suggest logging the exception here as an error, `offsetStart = offsetEnd;` isn't required

Reply:
This code is required for the REST API to function properly.   This is specific to the Polygon API and is the desired behavior to finish iterating through the pagination.

Reply:
if `isException` is `true` setting `offsetStart = offsetEnd` isn't required because it will go to the else case at line 623 and perform `offsetStart = offsetEnd`?

Reply:
Fixed